src/basic/chase.c

```diff
@@ -566,6 +566,11 @@ int chase_and_open(const char *path, const char *root, ChaseFlags chase_flags, i
 
         assert(!(chase_flags & (CHASE_NONEXISTENT|CHASE_STEP)));
 
+        if (FLAGS_SET(chase_flags, CHASE_PARENT))
+                open_flags |= O_DIRECTORY;
```

Comment:
I'm curious when this matters, if `CHASE_PARENT` is set we'll always just reopen the file descriptor returned by `chase()`, and `chase()` will already ensure that the returned file descriptor points to a directory, so neither the open flags nor the mode will have an effect at that point? Or maybe I'm missing something here.

Reply:
This is just for safety, and should not change any behavior. I extended the commit message.